### PR TITLE
Feat/refactor dev

### PR DESCRIPTION
**Pull request by Foivos**

### CODE REVIEW

File: pages/resources/supported.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Non-EVM Networks 
+
+- Axelarnet
+- cosmoshub
+- terra
+- crescent
+- juno
+- osmosis
+- osmosis-2
```
Comment (milapsheth) [inline pages/resources/supported.md:23]:
```suggestion
- Axelarnet
- terra
- osmosis-3
```
Others are outdated/frozen

File: pages/dev/gas-receiver.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    string calldata destinationChain,
+    string calldata destinationAddress,
+    bytes calldata payload,
+    string calldata symbol,
+    uint256 amount,
+    address refundAddress
+) external payable;
+```
+
+The function names are prety self explenatory. The following is true for the arguments:
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:76]: explenatory -> explanatory

File: pages/dev/gas-receiver.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+- For all functions
+  - `sender` needs to match the address that calls `callContrat` or `callContractWithToken` on the `AxelarGateway`. If the `AxelarGasReceiver` is called by the same contract that will call the gateway then simply specify `address(this)` as `sender`.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:79]: `callContrat` -> `callContract` I think.

File: pages/dev/gas-receiver.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+need to match the arguments of a `contractCall` on the `AxelarGateway`
+- For `payGasForContractCallWtihToken` and `payNativeGasForContractCallWithToken`
+  - `destinationChain`
+  - `destinationAddress`
+  - `payload`
+  - `symbol`
+  - `amount`
+need to match the arguments of a `contractCallWithToken` on the `AxelarGateway`
+- For `payGasForContractCall` and `payGasForContractCallWtihToken`
+  - `gasToken` is the address of the token that gas will be payed in. Ensure this token is supported with our API.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:93]: payed -> paid

File: pages/dev/gas-receiver.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  - `gasFeeAmount` is the amount of `gasToken` to trasnfer from the sender. The sender needs to have approved the `AxelarGasReceiver` with the appropriate amound to `gasToken` first.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:94]: trasnfer -> transfer

File: pages/dev/gas-receiver.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+- For `payNativeGasForContractCall` and `payNativeGasForContractCallWithToken` the amount of funds received is specified by `msg.value`.
+- For all functions
+  - `refundAddress` is the address that will be able to receive excess amount payed for gas.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:97]: payed -> paid

File: pages/dev/gas-receiver.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
+# Axelar Gas Receiver
+
+The Axelar Network only validates that a contract call was made from a certain source chain to another destination chain. Execution of the actual contract call is up to the application. We also provide a set of services that can automatically exeute the contract calls on the destination chain by receiving funds on the source chain.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:3]: Network -> network

File: pages/dev/gas-receiver.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+## Introduction
+
+An application that wants Axelar to automatically execute contract calls on the destination chain needs to do four things:
+
+- Estimate the `gasLimit` that the contract call will require on the destionation chain.
+- Query our API to get the `sourceTokenPrice` of the desired token that gas will be payed in, as well as `destinationGasPrice` and `destinationTokenPrice`, the price of the native token, on the destination chain.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:10]: payed -> paid

File: pages/dev/gas-receiver.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+- Calcualate the amount of token to be payed as `gasLimit * destinationGasPrice * gestinationTokenPrice / sourceTokenPrice`.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:11]: payed -> paid

File: pages/dev/gas-receiver.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+- Pay our `AxelarGasReceiver` smart contract on the source chain that amount. This can be done by the application's smart contracts so no additional transactions are required (except maybe approval in case gas is payed in non-native tokens).
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:12]: payed -> paid

File: pages/dev/gas-receiver.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+
+Our service does the following:
+
+- Monitors `AxelarGasReceiver` for receipts of payment, and gets the amount payed as `amountPayed`.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:16]: payed -> paid

File: pages/dev/gas-receiver.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+- Matches those to contract calls.
+- Queries our API to get the `sourceTokenPrice` of the token that gas was payed in, as well as `destinationGasPrice` and `destinationTokenPrice`, the price of the native token, on the destination chain.
+- Calcualate the `gasLimit` as `amountPayed * sourceTokenPrice / (destinationGasPrice * gestinationTokenPrice)`.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:19]: `amountPayed` really should be `amountPaid`, but that requires a change to the gas receiver itself, I am guessing.

File: pages/dev/gas-receiver.md [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+- Estimate the `gasLimit` that the contract call will require on the destionation chain.
+- Query our API to get the `sourceTokenPrice` of the desired token that gas will be payed in, as well as `destinationGasPrice` and `destinationTokenPrice`, the price of the native token, on the destination chain.
+- Calcualate the amount of token to be payed as `gasLimit * destinationGasPrice * gestinationTokenPrice / sourceTokenPrice`.
+- Pay our `AxelarGasReceiver` smart contract on the source chain that amount. This can be done by the application's smart contracts so no additional transactions are required (except maybe approval in case gas is payed in non-native tokens).
+
+Our service does the following:
+
+- Monitors `AxelarGasReceiver` for receipts of payment, and gets the amount payed as `amountPayed`.
+- Matches those to contract calls.
+- Queries our API to get the `sourceTokenPrice` of the token that gas was payed in, as well as `destinationGasPrice` and `destinationTokenPrice`, the price of the native token, on the destination chain.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:18]: payed -> paid

File: pages/dev/gas-receiver.md [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+- Calcualate the `gasLimit` as `amountPayed * sourceTokenPrice / (destinationGasPrice * gestinationTokenPrice)`.
+- Executes the specified contract call specifying a the `gasLimit` specified above.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:20]: Delete "a"

File: pages/dev/gas-receiver.md [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+
+We plan to add an option to get refunds in case excecive amounts are payed as gas, but this is not yet implemented.
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:22]: excecive -> excessive
payed -> paid

File: pages/dev/gas-receiver.md [hunk 16]
```diff
@@ -0,0 +1,134 @@
+# Axelar Gas Receiver
```
Comment (galenmoore) [inline pages/dev/gas-receiver.md:1]: Not sure where this is going on the outline, but I would suggest it should go beneath General Message Passing: 
# Developers
## Introduction
## Get started
## Transfer tokens
## General Message Passing
## Axelar Gas Receiver

File: pages/dev/get-started.md [hunk 17]
```diff
@@ -7,16 +7,11 @@ import Button from '../../components/button'
 1. **_Build._** Develop your dapp, test against the Axelar local development environment.
 2. **_Deploy._** Deploy your contracts, point your dapp to a live network: testnet or mainnet.
 
-## Example: Hello-world
+## Examples
 
-The `testnet` project is a complete, working example illustrating the build-deploy process.
+There are several examples at the `axelar-local-gmp-examples` project complete with instruction of how to run each on your local machine or testnet. View the github README for instructions and code:
```
Comment (galenmoore) [inline pages/dev/get-started.md:12]: instruction of -> instructions on

File: pages/dev/gas-receiver.md [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+  - `gasToken` is the address of the token that gas will be paid in. Ensure this token is supported with our API.
+  - `gasFeeAmount` is the amount of `gasToken` to transfer from the sender. The sender needs to have approved the `AxelarGasReceiver` with the appropriate amount to `gasToken` first.
+- For `payNativeGasForContractCall` and `payNativeGasForContractCallWithToken` the amount of funds received is specified by `msg.value`.
+- For all functions
+  - `refundAddress` is the address that will be able to receive excess amount paid for gas.
+
+
+## API
+
+To get the relative gas cost in any token on any of the supported chains you can use the following script. This is subject to change.
```
Comment (ggutoski) [inline pages/dev/gas-receiver.md:102]: We should provide an API call for this in the SDK.

**Review by ggutoski — COMMENTED** [review]
I'll push a few tweaks and then merge

**Review by ggutoski — APPROVED** [review]

File: pages/dev/get-started.md [hunk 19]
```diff
@@ -7,16 +7,11 @@ import Button from '../../components/button'
 1. **_Build._** Develop your dapp, test against the Axelar local development environment.
 2. **_Deploy._** Deploy your contracts, point your dapp to a live network: testnet or mainnet.
 
-## Example: Hello-world
+## Examples
 
-The `testnet` project is a complete, working example illustrating the build-deploy process.
+There are several examples at the `axelar-local-gmp-examples` project complete with instruction on how to run each on your local machine or testnet. View the github README for instructions and code:
 
-1. Run the dapp in the local development environment.
```
[the remark this answers is not in the document]
Comment (galenmoore) [reply pages/dev/get-started.md:14]: Doesn't this say the same thing as 1 & 2 above?

[the remark this answers is not in the document]
Comment (ggutoski) [reply pages/dev/get-started.md:14]: One could argue that 😛 .  It does clarify that the contracts are already deployed on testnet, and it does serve to drive the point home IMO.  But I can remove it.

**Pull request merged**